src/backend/tools/go.CLI/cmd/gocaliopen/cli_cmds/fixMissingParticipants.go

```diff
@@ -0,0 +1,192 @@
+	Run: fixMissingParticipants,
+}
+
+func init() {
+	RootCmd.AddCommand(fixMissingParticipantsCmd)
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// fixMissingParticipantsCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// fixMissingParticipantsCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+func fixMissingParticipants(cmd *cobra.Command, args []string) {
+	var err error
+	//check/get connexions to facilities
+	//store
+	var Store *store.CassandraBackend
+	Store, err = getStoreFacility()
+	if err != nil {
+		log.WithError(err).Fatalf("initialization of %s backend failed", apiConf.BackendName)
+	}
+	defer Store.Close()
+	//index
+	var Index *index.ElasticSearchBackend
+	Index, err = getIndexFacility()
+	if err != nil {
+		log.WithError(err).Fatalf("initialization of %s index failed", apiConf.IndexConfig.IndexName)
+	}
+	defer Index.Close()
+	//nats
+	var MsgQueue *nats.Conn
+	MsgQueue, err = getMsgSystemFacility()
+	if err != nil {
+		log.WithError(err).Fatal("initiazation of message queue failed")
+	}
+
+	var total int
+	count := 0
+	invalidCount := 0
+	fixedCount := 0
+	failedCount := 0
+	//
+	handleInvalidMessage := func(msgId, usrId, rawMsgId gocql.UUID) {
+		invalidCount++
+		//try to fetch message from index and check if it has participants
+		result, err := Index.Client.Get().Index(usrId.String()).Type(objects.MessageIndexType).Id(msgId.String()).Do(context.TODO())
+		if err != nil && !strings.Contains(err.Error(), "Not Found") {
+			log.Warn(err)
+			failedCount++
+			return
+		}
+		if result != nil && result.Found {
+			msg := new(objects.Message).NewEmpty().(*objects.Message)
+			if err := json.Unmarshal(*result.Source, msg); err != nil {
+				log.Warn(err)
+				failedCount++
+				return
+			}
+			err = msg.Message_id.UnmarshalBinary(msgId.Bytes())
+			if err != nil {
+				log.WithError(err).Warn("failed to unmarshal messageId")
+				failedCount++
+				return
+			}
+			if msg.Participants != nil && len(msg.Participants) > 0 {
+				err = Store.UpdateMessage(msg, map[string]interface{}{
+					"Participants": msg.Participants,
+				})
+				if err != nil {
+					log.WithError(err).Warnf("failed to update message in db")
+					failedCount++
+					return
+				}
+				fixedCount++
+				return
+			}
+		}
+		log.Infoln("failed to retrieve participant from index, trying to reinject raw message")
+		err = reInjectRaw(usrId.String(), rawMsgId.String(), MsgQueue)
+		if err != nil {
+			log.WithError(err).Warn("failed to re-inject raw message in stack")
+			failedCount++
+			return
+		}
+		//reinjection OK, delete former invalid message
+		err = Store.Session.Query(`DELETE FROM message WHERE message_id = ? AND user_id = ?`, msgId, usrId).Exec()
```

Comment:
there is some state on previous message that we will make disappear: read status at least